lib/rubocop/cop/rspec/rails/minitest_assertions.rb

```diff
@@ -211,17 +224,9 @@ def assertion
 
           RESTRICT_ON_SEND = ASSERTION_MATCHERS.flat_map { |m| m::MATCHERS }
 
-          ASSERTION_MATCHERS.each do |m|
-            name = m.name.split('::').last
-
-            def_node_matcher "minitest_#{name}".to_sym, m::NODE_MATCHER_PATTERN
-          end
-
           def on_send(node)
             ASSERTION_MATCHERS.each do |m|
-              name = m.name.split('::').last
-
-              public_send("minitest_#{name}".to_sym, node) do |*args|
+              m.match_assertion(node) do |*args|
```

Comment:
at this point the names are getting a little silly - I'll next explore if we can nicely have the classes handle calling their `match` logic and just return a result